dvc/stage/loader.py

```diff
@@ -106,13 +107,23 @@ def __getitem__(self, name):
                 "No lock entry found for '%s:%s'", self.dvcfile.relpath, name,
             )
 
-        return self.load_stage(
+        stage = self.load_stage(
             self.dvcfile,
             name,
             self.resolved_data[name],
             self.lockfile_data.get(name, {}),
         )
 
+        group, *keys = name.rsplit(JOIN, maxsplit=1)
+        if group and keys and name in self.stages_data:
+            stage.raw_data.generated_from = group
```

Comment:
It seems to me that in the test case `build@us` `generated_from` is not set - shouldn't it be set to `build`?

Reply:
Silly mistake. It should have been
```suggestion
        if group and keys and name not in self.stages_data:
            stage.raw_data.generated_from = group
```